File: php/LinkedList.php
<?php

class Node {
    public $value = null;
    public $next = null;

    public function __construct($value){
        $this->value = $value;
    }
}

class LinkedList {
    public $root = null;

    public function add_last($node){
        if(is_null($this->root)){
            $this->root = $node;
        } else {
            $parent = $this->root;
            while(!is_null($parent->next)){
                $parent = $parent->next;
            }
            $parent->next = $node;
        }
    }

    public function __toString(){
        if(is_null($this->root)) {
            return "Empty list\n";
        } else {
            $output = $this->root->value;
            $node = $this->root->next;
            while(!is_null($node)){
                $output .= ' -> ' . $node->value;
                $node = $node->next;
            }
            return $output;
        }
    }
}

/* Test
$n = new Node(1);
$ll = new LinkedList();
$ll->add_last($n);
$n = new Node(2);
$ll->add_last($n);
$n = new Node(4);
$ll->add_last($n);
$n = new Node(3);
$ll->add_last($n);
echo $ll;
*/

File: python/LinkedList.py
class Node(object):
    value = None
    next = None

    def __init__(self, value):
        self.value = value

    def __repr__(self):
        return '<Node: %s>' % self.value


class LinkedList(object):
    root = None

    def append(self, node):
        if not self.root:
            self.root = node
        else:
            parent = self.root
            while parent.next:
                parent = parent.next
            parent.next = node

    def __repr__(self):
        nodes = []
        node = self.root
        while node:
            nodes.append(str(node.value))
            node = node.next
        return '<Linked List: [%s]>' % ' -> '.join(nodes)


### Test
#if __name__ == '__main__':
#    ll = LinkedList()
#    ll.append(Node(1))
#    ll.append(Node(2))
#    ll.append(Node(3))
#    ll.append(Node(4))
#    ll.append(Node(5))
#    print ll
